File: py_version/schemas.py
from fastapi import UploadFile
from pydantic import BaseModel, Field, ConfigDict
from typing import Optional, List, Dict, Any
from dataclasses import dataclass
from enums import InputType, LLMProvider
from tools.base import BaseTool, Tool


class CallAgentRequest(BaseModel):
    request_text: str
    agent_names: List[str]
    request_files: Optional[List[UploadFile]] = None

class CallAgentResponse(BaseModel):
    request_text: str
    request_files: Optional[List[str]] = None
    result: str
    status: int = 200


class Agent(BaseModel):
    model_config = ConfigDict(arbitrary_types_allowed=True)
    system_prompt: str
    name: str
    tools: List[BaseTool]
    llm_provider: str
    model: str


# OpenAI-related schemas
class ToolCall(BaseModel):
    id: str
    name: str
    arguments: Dict[str, Any]
    output: Optional[str] = None


class OpenAIResponse(BaseModel):
    content: str
    tool_calls: List[ToolCall] = Field(default_factory=list)
    total_tokens: Optional[int] = None
    model_used: str


class AgentDefinition(BaseModel):
    name: str
    system_prompt: str
    tools: List[Tool]
    llm_provider: str
    model: str

# Temporal workflow schemas
@dataclass
class AgentRequest:
    request_text: str
    request_files: List[str]
    agents: List[AgentDefinition]

File: ts_version/src/types/schemas.ts
// TypeScript equivalents of Pydantic models from schemas.py
import { InputType, LLMProvider } from './enums';

// Tool-related interfaces (from tools/base.py)
export interface ToolInput {
  name: string;
  type: InputType;
  description: string;
  required: boolean;
}

export interface Tool {
  name: string;
  description: string;
  inputs: ToolInput[];
}

// Abstract base for callable tools
export abstract class BaseTool {
  abstract name: string;
  abstract description: string;
  abstract inputs: ToolInput[];
  
  abstract call(kwargs: Record<string, any>): Promise<string>;
}

// FastAPI request/response models
export interface CallAgentRequest {
  request_text: string;
  agent_names: string[];
  request_files?: File[];
}

export interface CallAgentResponse {
  request_text: string;
  request_files?: string[];
  result: string;
  status: number;
}

// Agent-related interfaces
export interface Agent {
  system_prompt: string;
  name: string;
  tools: BaseTool[];
  llm_provider: string;
  model: string;
}

export interface AgentDefinition {
  name: string;
  system_prompt: string;
  tools: Tool[];
  llm_provider: string;
  model: string;
}

// OpenAI-related schemas
export interface ToolCall {
  id: string;
  name: string;
  arguments: Record<string, any>;
  output?: string;
}

export interface OpenAIResponse {
  content: string;
  tool_calls: ToolCall[];
  total_tokens?: number;
  model_used: string;
}

// Temporal workflow schemas (from dataclass)
export interface AgentRequest {
  request_text: string;
  request_files: string[];
  agents: AgentDefinition[];
}